module_auto_update/data/cron_data.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <record model="ir.cron" id="module_check_upgrades_cron">
+        <field name="name">Check for Module Upgrades</field>
+        <field name="active" eval="True"/>
+        <field name="user_id" ref="base.user_root"/>
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+        <field name="numbercall">-1</field>
+        <field name="nextcall" eval="(DateTime.now() + timedelta(days= +1)).strftime('%Y-%m-%d 3:00:00')"/>
+        <field name="model">ir.module.module</field>
+        <field name="function">check_upgrade_availability</field>
+        <field name="args" eval="'()'"/>
+    </record>
+
+    <record model="ir.cron" id="base_module_upgrade_cron">
+        <field name="name">Perform Scheduled Module Upgrades</field>
```

Comment:
I see no sense on having 2 crons. Checking the upgrades without upgrading makes no sense, as upgrading without checking what's to upgrade neither. Besides, you could have a race condition where there are more than 1 cron workers. Please have a Python method that does all and run it in a single cron.